tests/parametric/test_otel_span_methods.py

```diff
@@ -286,3 +286,608 @@ def test_otel_get_span_context(self, test_agent, test_library):
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "go", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_operation_name_aws_client(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `rpc.system + "." + rpc.service.ToLower()` :
+            - Span kind is set to Client
+            - rpc.system is set to aws-api
+            - rpc.service is set to something (note the ToLower() on it when making the operation name)
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/instrumentation/aws-sdk/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_CLIENT) as span:
+                span.set_attributes({"rpc.system": "aws-api"})
+                span.set_attributes({"rpc.service": "S3"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, OtelSpan(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "aws.s3"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "go", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @irrelevant(context.library == "cpp", reason="library does not implement OpenTelemetry")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    def test_otel_span_operation_name_rpc_client(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to the `rpc.system + ".request"`
+            - Span kind is set to Client
+            - rpc.system is set to something
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/rpc/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_CLIENT) as span:
+                span.set_attributes({"rpc.system": "grpc"})
```

Comment:
@dougqh @amarziali I'll restart this comment thread here, but should we special-case the RPC logic when `rpc.system == grpc`? Our v1 operation naming effort has standardized on operation names `grpc.client.request` and `grpc.server.request` so I wonder if we should do the same in our OTEL logic. WDYT?

Reply:
I'm on board with the general case BTW, so the existing logic in this method looks fine for other values of `rpc.system`

Reply:
I've updated these to follow this (and per the comments on the doc): `rpc.system` + "." + `span.kind` + "." + `"request"`

Reply:
The only "special" `rpc.system` is for `aws-api`, the others will use `rpc.system` + "." + `span.kind` + "." + "request"